src-tauri/src/state/reducers.rs

```diff
@@ -0,0 +1,111 @@
+use crate::state::actions::Action;
+use crate::state::state::{AppState, Profile, StateStatus};
+use tracing::{warn, info};
+
+/// Sets the locale to the given value. If the locale is not supported yet, the current locale will stay unchanged.
+pub fn set_locale(state: &AppState, action: Action) -> anyhow::Result<()> {
+    let payload = action.payload.unwrap();
```

Comment:
I suggest to use `Option::ok_or` to avoid panicking on invalid input an return an `Result::Err()` instead.
```suggestion
    let payload = action.payload.ok_or(anyhow::anyhow!("unable to read payload"))?;
```